cfme/fixtures/cli.py

```diff
@@ -249,40 +251,184 @@ def ha_appliances_with_providers(ha_multiple_preupdate_appliances, app_creds):
     app0_ip = apps0.hostname
     app1_ip = apps1.hostname
     pwd = app_creds["password"]
+
+    def logging_callback(appliance):
+        def the_logger(m):
+            logger.debug('Appliance %s:\n%s', appliance.hostname, m)
+        return the_logger
+
     # Configure first appliance as dedicated database
-    command_set = ("ap", "", "5", "1", "1", "1", "y", pwd, TimedCommand(pwd, 360), "")
-    apps0.appliance_console.run_commands(command_set)
-    wait_for(lambda: apps0.db.is_dedicated_active)
+    interaction = SSHClientInteraction(apps0.ssh_client, timeout=10, display=True,
+                                       output_callback=logging_callback(apps0))
+    interaction.send('ap')
```

Comment:
Question: Is there a way to reduce LOC here, e.g. by passing a list of commands to run on the appliance. I am not a fan of the lengthy calls to `expect` and `send`, especially with all these hard-coded strings. What is your reasoning for implementing this over the old method?

Reply:
The old method is just blindly sending answers without checking the questions. The question change and then it is hard to debug why things gone bad.

Reply:
@jaryn My thought is that this should be a method of the the `SSHClient` class. You could define a method `interact` that takes two lists, `send` and `expect`, and allowing for the `timeout` kwarg maybe via a `namedtuple`? Then you can call interact and give it these lists of commands. What do you think about this?

Reply:
@jaryn What are your thoughts on this? I still this that it would be good to wrap these interaction calls in a method.

Reply:
@jaryn This comment still applies

Reply:
@john-dupuy  I am not sure I understand how do you think it should work. If I do understand that you want me to specify some structure of strings to expect and send and some function which would take the structure and executed the commands. That could be done, but I think I like better the way how it is now. It is easy to find on which point the code did break as one can easy see that in the debugger and a console log of pytests. Easy debuggability is one point why did I introduce using of pyexpect.

I started working on other approach how to solve the problem with lengthy procedures:
PR #8137/

Reply:
@jaryn What I'm envisioning is something like,

```python,
def ssh_interact(ssh_interaction, send, expect):
    """
    Send commands via ssh, and check that the expected result of the command appears.

    send: list of tuples containing any number of send commands
    expect: list of dicts contain expected outcomes and timeout kwarg
            e.g. expect = [ {"expected": blah blah blah, "timeout": 20} ]
    ssh_interaction: SSHClientInteraction instance
    """
    if len(send) != len(expect):
        raise ValueError("Length of send list must match that of expect list!")

    for cmds, result in zip(send, expect):
        for s in cmds:
            ssh_interaction.send(s)
        if result["timeout"]:
            ssh_interaction.expect(result["expected"], timeout=result["timeout"])
        else:
            ssh_interaction.expect(result["expected"])
```
This may not be the best way to do it, but I would prefer to have these interaction statements put into a method with the list of commands defined as a constant list. What do you think?

Reply:
@jaryn here i agree with @john-dupuy it will better to create on definition.

Reply:
@john-dupuy @digitronik maybe there would be less lines of code but i don't think this justifies worse debugability. When we are doing sequence of calls, and an exception happens, the pdb will stop on the exact line where things went wrong and it is then obvious which string was sent. While when done in the fashion as @john-dupuy suggest, then one needs to find what was sent by displaying `s`, then find that line in the list of dicts, then he can start debugging.

And when debugging, with my stupid sipmle approach,  I can just copy'n'paste the lines from editor to debgger to execute them. While when done in your way I would have to modify them.

Disk space is dead cheap. Human work is not.

Reply:
I think I managed to decrease the LOC while keeping the code simple and debuggable.

Reply:
@jaryn I understand your concern about debugging, but there are ways to bake that into the method. For example, you could log the commands and expected results as they're sent with `logger.debug`. You could have the exception print out the expected vs received result as well. 

So I think there are ways to get the debugging you want while also writing a general method that will be useful elsewhere and make the code easier to maintain down the line.